### PR TITLE
[CUMULUS-483] Backup and restore documentation

### PR DESCRIPTION
**Pull request by scisco**
- adds backup and restore documentation
- removes all the build artifacts that were added to the `develop` branch because `master` was to merged it
- adds more ignore files to ensure build artifacts are not tracked
- reorganizes the docs folder to make it less cluttered
- improves the main repo's readme

### CODE REVIEW

File: docs/backup_and_restore.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+# Backup and Restore
+
+Cumulus-api uses a number of methods to preserve the metadata generated in a Cumulus instance.
```
Comment (abarciauskas-bgse) [inline docs/backup_and_restore.md]: Nit but I prefer to call this by the npm module name `@cumulus/api`

File: docs/backup_and_restore.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+## Data in DynamoDB
+
+All configurations and system generated metadata is stored in DynamoDB tables except the logs. System logs are stored in the AWS CloudWatch service.
```
Comment (abarciauskas-bgse) [inline docs/backup_and_restore.md]: `system-generated metadata`

File: docs/backup_and_restore.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+Cumulus-api creates the following DynamoDB tables:
+
+- **Users:** list of the api/dashboard users
```
Comment (abarciauskas-bgse) [inline docs/backup_and_restore.md]: Minor but do we need to prefix these descriptions with "list of " ?

```
 **Collections:** collection records 
**Providers:** provider records
**Rules:** rules for managing and running workflows
**Executions:** workflow executions (step function executions)
**Granules:** granules processed by the Cumulus instance
**PDRs:** PDRs processed in Cumulus
```

File: docs/backup_and_restore.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+On-demand backups allow you to create backups of DynamoDB table data and its settings. You can initiate an on-demand backup at any time with a single click from the AWS Management Console or a single API call. You can restore the backups to a new DynamoDB table in the same AWS Region at any time.
+
+PITR gives your DynamoDB tables continuous protection from accidental writes and deletes. With PITR, you do not have to worry about creating, maintaining, or scheduling backups. You enable PITR on your table and your backup is available for restore at any point in time from the moment you enable it, up to a maximum of the 35 preceding days. For example, imagine a test script writing accidentally to a production DynamoDB table. You could recover your table to any point in time within the last 35 days.
+
+On-demand backups help with long-term archival requirements for regulatory compliance. On-demand backups give you full-control of managing the lifecycle of your backups, from creating as many backups as you need to retaining these for as long as you need.
```
Comment (abarciauskas-bgse) [inline docs/backup_and_restore.md]: Are backups preserved when tables are destroyed? (God forbid someone destroys a stack 😱 )

File: docs/backup_and_restore.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+You can enable point-in-time recovery on all existing tables in your `config.yml`. Add the following configuration to your `config.yml` under a deployment section:
+
+```yaml
+default:
+
+    enablePointInTime: true
+```
+
+**Imoprtant Note:** Configuring Point-in-time recovery is not supported by the CloudFormation (as of June 2018). We enable this feature deployment using AWS API. However, due to a limitation of AWS API, the feature fails to be enabled if it is running against newly created tables.
```
Comment (abarciauskas-bgse) [inline docs/backup_and_restore.md]: Nit: lowercase `point-in-time` (or use the acronym?)

Comment (abarciauskas-bgse) [reply docs/backup_and_restore.md]: This is interesting - so if you try to use PITR on a new stack it won't work?

Comment (abarciauskas-bgse) [reply docs/backup_and_restore.md]: Oh sorry you answered this in the next line 🤦‍♂️

**Review by abarciauskas-bgse — APPROVED** [review]
Made some minor comments but looks great overall 💯

File: docs/backup_and_restore.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+Therefore, if you are deploying a new stack, make sure the feature is turned off on your first deployment. You can turn it on and enable about an hour after your tables are created.
+
+
+## Backup and Restore with cumulus-api CLI
+
+cumulus-api CLI also includes a backup and restore command. The backup command, download the content of any of your DynamoDB tables to single `.json` files. You can use these `.json` files to upload the the records to another DynamoDB table.
```
Comment (laurenfrederick) [inline docs/backup_and_restore.md]: 'The backup command,' should be something like 'To use the backup command, '

Comment (laurenfrederick) [reply docs/backup_and_restore.md]: How do you download the content of a table to a JSON? Is that really straightfoward?

**Review by laurenfrederick — COMMENTED** [review]
Small changes

File: docs/backup_and_restore.md (hunk 4 above)
Comment (scisco) [reply docs/backup_and_restore.md]: Good question. I don't know

**Pull request merged**